R/dplyr.R

```diff
@@ -279,6 +298,16 @@ summarise.zoomed_dm <- function(.data, ...) {
   replace_zoomed_tbl(.data, summarized_tbl, new_tracked_cols_zoom)
 }
 
+
+#' @rdname dplyr_table_manipulation
+#' @export
+summarise.dm_keyed_tbl <- function(.data, ...) {
+  keys_info <- keyed_get_info(.data)
+  tbl <- unclass_keyed_tbl(.data)
+  summarised_tbl <- summarise(tbl, ...)
+  new_keyed_tbl_from_keys_info(summarised_tbl, keys_info)
```

Comment:
The grouping columns become primary keys here.

Reply:
After `group_by()` + `summarize()`, the `tzone` column should be the new primary key. (I'm aware that `summarize()` can create more than one row these days, I think this is a mistake. We'll still assume the default case and can check key properties later.)

Reply:
https://twitter.com/cynkra_com/status/1532806522551652357 : `flights_agg` is the result of `flights %>% count(origin)` .

Reply:
Like so? 

``` r
library(dm, warn.conflicts = FALSE)

dm <- dm_nycflights13()

dm$airports %>% 
  summarise(mean_alt = mean(alt))
#> # A tibble: 1 x 1
#> # Keys:     --- | 1 | 0
#>   mean_alt
#> *    <dbl>
#> 1     632.

dm$airports %>% 
  group_by(tzone, dst) %>%
  summarise(mean_alt = mean(alt))
#> # A tibble: 6 x 3
#> # Groups:   tzone [6]
#> # Keys:     `tzone` | 1 | 0
#>   tzone               dst   mean_alt
#> * <chr>               <chr>    <dbl>
#> 1 America/Chicago     A         680.
#> 2 America/Denver      A        5399.
#> 3 America/Los_Angeles A         313.
#> 4 America/New_York    A         396.
#> 5 America/Phoenix     N        1135 
#> 6 Pacific/Honolulu    N          13
```

<sup>Created on 2022-07-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

Few questions:

- We should always leave out foreign keys here, correct?
- The resulting table from dplyr/tidyr operations should get a new UUID, right? I am assuming that (just like `flights_agg`), the table resulting from these operations is expected to be sometimes added back to the original `dm` object?

Reply:
- Leave out foreign keys
- Not sure about a new UUID, we can start with that (because it's safer) and reuse the existing later if we find it's better